libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs

```diff
@@ -614,7 +635,6 @@ private async Task<DialogTurnResult> CheckForMultiTurnPromptAsync(WaterfallStepC
                 if (answer.Context != null && answer.Context.Prompts.Count() > 0)
                 {
                     var previousContextData = ObjectPath.GetPathValue(stepContext.ActiveDialog.State, QnAContextData, new Dictionary<string, int>());
```

Comment:
What's the point of passing in a new dictionary, if you never look at the return values that come back? The design of the API seems is may need to be revisited.

Reply:
The  new Dictionary is being used as default value, in case the expected value is not found.  
Snippet for quick reference.

/// <summary>
        /// Get the value for a path relative to an object.
        /// </summary>
        /// <typeparam name="T">type to return.</typeparam>
        /// <param name="obj">object to start with.</param>
        /// <param name="path">path to evaluate.</param>
        /// <param name="defaultValue">default value to use if any part of the path is missing.</param>
        /// <returns>value or default(T).</returns>
        public static T GetPathValue<T>(object obj, string path, T defaultValue)
        {
            if (TryGetPathValue<T>(obj, path, out var value))
            {
                return value;
            }

            return defaultValue;
        }


---
In reply to: [430549913](https://github.com/microsoft/botbuilder-dotnet/pull/3935#discussion_r430549913) [](ancestors = 430549913)